js/adapt-youtube.js

```diff
@@ -1,208 +1,206 @@
+
+    if (window.onYouTubeIframeAPIReady !== undefined) return;
+    window.onYouTubeIframeAPIReady = () => {
+      Adapt.log.info('YouTube iframe API loaded');
+      Adapt.youTubeIframeAPIReady = true;
+      Adapt.trigger('youTubeIframeAPIReady');
+    };
+    $.getScript('//www.youtube.com/iframe_api');
+  }
+
+  preRender() {
+    this.listenTo(Adapt, {
+      'device:resize': this.setIFrameSize,
+      'device:changed': this.setIFrameSize,
+      'media:stop': this.onMediaStop
+    });
+  }
+
+  setIFrameSize() {
+    const $iframe = this.$('iframe');
+    const widgetWidth = this.$('.component__widget').width();
+
+    $iframe.width(widgetWidth);
+
+    // default aspect ratio to 16:9 if not specified
+    const aspectRatio = this.model.get('_media')._aspectRatio 
+      ? parseFloat(this.model.get('_media')._aspectRatio)
+      : 1.778;
+    if (isNaN(aspectRatio)) return;
+    $iframe.height(widgetWidth / aspectRatio);
+  }
+
+  postRender() {
+    //FOR HTML/HBS Paramenters: https://developers.google.com/youtube/player_parameters
+    if (Adapt.youTubeIframeAPIReady === true) {
+      this.onYouTubeIframeAPIReady();
+      return;
+    }
+    Adapt.once('youTubeIframeAPIReady', this.onYouTubeIframeAPIReady, this);
+  }
 
-      if (window.onYouTubeIframeAPIReady === undefined) {
-        window.onYouTubeIframeAPIReady = function() {
-          Adapt.log.info('YouTube iframe API loaded');
-          Adapt.youTubeIframeAPIReady = true;
-          Adapt.trigger('youTubeIframeAPIReady');
-        };
-        $.getScript('//www.youtube.com/iframe_api');
-      }
-    },
+  remove() {
+    if (this.player !== null) {
+      this.player.destroy();
+    }
 
-    preRender: function() {
-      this.listenTo(Adapt, {
-        'device:resize': this.setIFrameSize,
-        'device:changed': this.setIFrameSize,
-        'media:stop': this.onMediaStop
-      });
-    },
+    super.remove();
+  }
 
-    setIFrameSize: function () {
-      var $iframe = this.$('iframe');
-      var widgetWidth = this.$('.component__widget').width();
+  setupEventListeners() {
+    this.completionEvent = (this.model.get('_setCompletionOn') || 'play');
 
-      $iframe.width(widgetWidth);
+    if (this.completionEvent !== 'inview') return;
+    this.setupInviewCompletion('.component__widget');
+  }
 
-      var aspectRatio = (this.model.get('_media')._aspectRatio ? parseFloat(this.model.get('_media')._aspectRatio) : 1.778);//default to 16:9 if not specified
-      if (!isNaN(aspectRatio)) {
-        $iframe.height(widgetWidth / aspectRatio);
-      }
-    },
-
-    postRender: function() {
-      //FOR HTML/HBS Paramenters: https://developers.google.com/youtube/player_parameters
-      if (Adapt.youTubeIframeAPIReady === true) {
-        this.onYouTubeIframeAPIReady();
-      } else {
-        Adapt.once('youTubeIframeAPIReady', this.onYouTubeIframeAPIReady, this);
+  onYouTubeIframeAPIReady() {
+    this.player = new YT.Player(this.$('iframe').get(0), {
```

Comment:
ESLint: undefined global.
```suggestion
    this.player = new window.YT.Player(this.$('iframe').get(0), {
```
See also other instances in `onPlayerStateChange()`.